test/e2e/namespace.go

```diff
@@ -124,9 +162,15 @@ func ensurePodsAreRemovedWhenNamespaceIsDeleted(f *framework.Framework) {
 			return false, nil
 		}))
 
-	By("Verifying there is no pod in the namespace")
+	By("Recreating the namespace")
+	namespace, err = f.CreateNamespace("nsdeletetest", nil)
```

Comment:
Not necessary today, id prefer to future proof.  Ie someone changes how namespace protections work.  This test shouldn't depend on us allowing you to access things outside a namespace